server/src/main/java/org/elasticsearch/common/util/concurrent/ThreadContext.java

```diff
@@ -657,12 +651,60 @@ private void writeTo(StreamOutput out, Map<String, String> defaultHeaders) throw
 
             out.writeMap(responseHeaders, StreamOutput::writeString, StreamOutput::writeStringCollection);
         }
+
+        boolean hasTracingIdInHeaders() {
+            return (requestHeaders.containsKey(Task.X_OPAQUE_ID) || requestHeaders.containsKey(Task.TRACE_ID));
+        }
+
+        Map<String, String> tracingHeaders() {
+            Map<String, String> result = new HashMap<>(2, 1);
+            if (requestHeaders.containsKey(Task.X_OPAQUE_ID)) {
+                result.put(Task.X_OPAQUE_ID, requestHeaders.get(Task.X_OPAQUE_ID));
+            }
+            if (requestHeaders.containsKey(Task.TRACE_ID)) {
+                result.put(Task.TRACE_ID, requestHeaders.get(Task.TRACE_ID));
+            }
+
+            return result;
+        }
+    }
+
+    public interface WithTracingIdsInThreadName {
```

Comment:
You are right, these methods are effectively static methods. I used an interface with default methods to encapsulate the two 'static' methods for use only in the wrapped runnables, since they rely logically on the second undoing the naming of the first.